source/simulator/solver.cc

```diff
@@ -790,10 +791,15 @@ namespace aspect
                 // output solver history
                 std::ofstream f((parameters.output_directory+"solver_history.txt").c_str());
 
-                for (unsigned int i=0; i<solver_control_cheap.get_history_data().size(); ++i)
-                  f << i << " " << solver_control_cheap.get_history_data()[i] << "\n";
+                if (parameters.n_cheap_stokes_solver_steps > 0)
```

Comment:
I see. Thanks. Maybe put a comment like "Only request the solver history if a history has actually been created" in front of the `if`?